singularity-eos/eos/eos_base.hpp

```diff
@@ -108,13 +141,27 @@ class EosBase {
               copy.TemperatureFromDensityInternalEnergy(rhos[i], sies[i], lambdas[i]);
         });
   }
-  template <typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer>
+  template <typename RealIndexer, typename ConstRealIndexer, typename LambdaIndexer,
+            typename = std::enable_if_t<!is_raw_pointer<RealIndexer, Real>::value>>
   inline void
   TemperatureFromDensityInternalEnergy(ConstRealIndexer &&rhos, ConstRealIndexer &&sies,
                                        RealIndexer &&temperatures, Real * /*scratch*/,
                                        const int num, LambdaIndexer &&lambdas) const {
-    TemperatureFromDensityInternalEnergy(rhos, sies, temperatures, num, lambdas);
+    TemperatureFromDensityInternalEnergy(std::forward<ConstRealIndexer>(rhos),
+                                         std::forward<ConstRealIndexer>(sies),
+                                         std::forward<RealIndexer>(temperatures), num,
+                                         std::forward<LambdaIndexer>(lambdas));
   }
+
+  template <typename LambdaIndexer>
+  inline void TemperatureFromDensityInternalEnergy(const Real *rhos, const Real *sies,
+                                                   Real *temperatures, Real * /*scratch*/,
+                                                   const int num, LambdaIndexer &&lambdas,
+                                                   Transform && = Transform()) const {
+    TemperatureFromDensityInternalEnergy(rhos, sies, temperatures, num,
+                                         std::forward<LambdaIndexer>(lambdas));
+  }
```

Comment:
This is clever---you let the underlying EOS do the transforms if it wants to, rather than putting that onus on the modifiers. 

As long as eospac is the special case I think this strategy makes a lot of sense. If we ever implement additional EOS's that we want to passthroughs with, then we may need to rethink. (And perhaps it's worth noting that somewhere.) But for now this is good. :+1: